diskv.go

```diff
@@ -38,14 +45,38 @@ var (
 // the final location of the data file will be <basedir>/ab/cde/f/abcdef
 type TransformFunction func(s string) []string
 
+// AdvancedTransformFunction transforms a key into:
+// * A slice of strings, (Path) with each element in the slice
+// representing a directory in the file path where the
+// key's entry will eventually be stored.
+//
+// and:
+//
+// * the file name
+//
+// For example, if TransformFunc transforms "abcdef/file.txt" to Path=["ab", "cde", "f"],
+// and FileName="file.txt"
+// the final location of the data file will be <basedir>/ab/cde/f/file.txt
+//
+// You must provide an InverseTransformFunction if you use this
+//
```

Comment:
This will look pretty weird in godoc. Please use regular punctuation, grammar, and formatting. e.g.

```go
// AdvancedTransformFunction transforms a key into a PathKey.
//
// A PathKey contains a slice of strings, where each element in the slice
// represents a directory in the file path where the key's entry will eventually
// be stored, as well as the filename.
//
// For example, if AdvancedTransformFunc transforms "abcdef/file.txt" to the
// PathKey {Path: ["ab", "cde", "f], FileName: "file.txt"}, the final location
// of the data file will be <basedir>/ab/cde/f/file.txt.
//
// You must provide an InverseTransformFunction if you use an
// AdvancedTransformFunction.
type AdvancedTransformFunction func(string) *PathKey
```